test/annotations_test.py

```diff
@@ -0,0 +1,45 @@
+import typing
+
+import pytest
+
+import synchronicity
+from synchronicity import Interface
+from synchronicity import async_wrap
+
+
+class _Bar:
+    pass
+
+class _Foo:
+    baz: _Bar
+    async def bar(self, arg1: typing.AsyncIterator[str]):
+        pass
+
+    @async_wrap.asynccontextmanager  # tests the annotations-compatible context manager factory
```

Comment:
why do we need to use a special asynccontextmanager here an not just the contextlib one?

Reply:
The problem is that the contextlib one doesn't transform (runtime) annotations on the returned object, so a correctly annotated (AsyncGenerator) underlying function wrapped by @contextlib.asynccontextmanager would get the AsyncGenerator type instead of AsyncContextManager when inspected :(

You don't strictly need it, if you are ok with "incorrectly" annotating the _underlying_ function as an AsyncContextManager instead of an AsyncGenerator.

Reply:
ok, that makes sense